FWCore/Framework/src/Schedule.cc

```diff
@@ -243,7 +251,51 @@ namespace edm {
             std::string friendlyClassName = pset.getParameter<std::string>("type");
             std::string productInstanceName = pset.getParameter<std::string>("fromProductInstance");
             std::string instanceAlias = pset.getParameter<std::string>("toProductInstance");
-            if (productInstanceName == star) {
+
+            if (friendlyClassName == star) {
+              bool match = false;
+              for (auto it = moduleLabelToBranches.lower_bound(moduleLabel);
+                   it != moduleLabelToBranches.end() && it->first == moduleLabel;
+                   ++it) {
+                if (productInstanceName != star and productInstanceName != it->second.productInstanceName()) {
+                  continue;
+                }
+                match = true;
+
+                checkAndInsertAlias(it->second.friendlyClassName(),
+                                    moduleLabel,
+                                    it->second.productInstanceName(),
+                                    processName,
+                                    alias,
+                                    instanceAlias,
+                                    preg,
+                                    aliasMap,
+                                    aliasKeys);
+              }
+              if (not match) {
+                // No product was found matching the alias.
+                // We throw an exception only if a module with the specified module label was created in this process.
+                for (auto const& product : preg.productList()) {
+                  if (moduleLabel == product.first.moduleLabel() && processName == product.first.processName()) {
+                    if (productInstanceName == star) {
+                      // This one doesn't really throw currently,
+                      // because there is no way a module would
+                      // produce something but would not be found with
+                      // wildcards for both the type and the instance
+                      // name. I'm leaving the exception here though
+                      // in case the condition above to throw an
+                      // exception gets changed.
+                      throw Exception(errors::Configuration, "EDAlias parameter set mismatch\n")
+                          << "There are no products with module label '" << moduleLabel << "'.\n";
```

Comment:
I would probably also delete this throw if it is logically impossible to get here. It is easy to add if we ever need it.

Reply:
I ended up throwing just throwing a single exception without checking `productInstanceName` (that way we'll at least get an indication if the logic was or becomes flawed).